presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -494,6 +494,20 @@ public static void validateParameters(Statement node, List<Expression> parameter
         }
     }
 
+    @Override
+    public void failQuery(QueryId queryId, Throwable cause)
+    {
+        requireNonNull(queryId, "queryId is null");
+        requireNonNull(cause, "cause is null");
+
+        log.warn(cause, "Fail query %s", queryId);
```

Comment:
Query %s failed. Also, I'm pretty sure we already log failures somewhere? Or if we don't, that there's a good reason for that.

Reply:
I see `QueryStateMachine` has `log.debug(throwable, "Query %s failed", queryId);` and yes we can drop this.